src/lib/utils/assert.hpp

```diff
@@ -42,18 +42,25 @@
 
 namespace opossum {
 
-[[noreturn]] inline void Fail(const std::string& msg) { throw std::logic_error(msg); }
+// We need this indirection so that we can throw exceptions from destructors without the compiler complaining. That is
+// generally forbidden and might lead to std::terminate, but since we don't want to handle most errors anyway,
+// that's fine.
+[[noreturn]] inline void fail(const std::string& msg) { throw std::logic_error(msg); }
```

Comment:
minor, but can we put this into `namespace detail` so people are less inclined to use fail() directly over Fail()?

Reply:
done